### ISSUE TITLE
doubt and update

### ISSUE THREAD

**Issue by ricoms** [GoogleCloudPlatform/tensorflow-without-a-phd#30]
Hi,

why in the `mnist_4.2_batchnorm_convolutional.py` file the biases are defined as constants? They are defined like this: `B5 = tf.Variable(tf.constant(0.1, tf.float32, [10]))` instead of the usual `tf.Variable(tf.ones([10])/10)`.

And also, I refactored all your mnist training to tensorflow estimator format, you can find it here: https://github.com/ricoms/tensorflow-without-a-phd
If you want I can create a pull request, although I deleted some of your files for visualization. For visualization, I am using a google Colaboratory ipynb that creates a tensorboard temporary url.

Another point is: my `mnist_4.2_batchnorm_convolutional.py` did not reach 99,5% accuracy. It's still below the `mnist_3.2` results:

![screenshot from 2018-10-10 08-33-22](https://user-images.githubusercontent.com/5393732/46733825-c2ff2900-cc67-11e8-8f58-1f739dd2ee71.png)

And I wonder if that is related to my first question, or my use of tf.layers.batch_normalization instead of your hardcoded function.

I intend to use it for a workshop at my college if it's ok with you. I will reference your work, of course.
@martin-gorner and team, thanks for all the great material and tutorial videos! I'm a fan.

**Comment by martin-gorner** [comment]
tf.ones/10 is a constant = 0.1
tf.constant(0.1) does the same thing.

Yes please do use this for your college teaching. That is the intended purpose of this material. Hack, modify and reshare at will. Thank you for the attribution.

This sample is not using the Estimator interface on purpose, because for real beginners, I wanted them to manipulate matrices and see what was going on and also because they usually like the animated training curves. I was planning to make the move to higher-level APIs but I have to figure out how to preserve some of the eye-candy first.

The next step is definitely to learn how to use Estimator and/or Keras. I would advise for Keras today. It has a cleaner object-oriented API.

Many things can go wrong with batch norm. I would venture that batch norm was not being applied in your version. Read more about it here: https://github.com/GoogleCloudPlatform/tensorflow-without-a-phd/blob/master/tensorflow-mnist-tutorial/README_BATCHNORM.md

**Comment by ricoms** [comment]
Thanks a lot for the clarification, tips and sharing your material.

You are right about the animated training curves. My approach was not good, the university firewall blocked the ngrock I used from Colaboratory. So we could not visualize Tensorboard. 0o'
But the workshop was a success, I think. :)

I will try something different for the visualization, maybe something closer to what you do. 
Also, I'm working on a different MNIST training, using the MNIST-Fashion dataset.

Thanks a lot again for all your great work!
